### PR TITLE
RDCC-5662: Location : Court Venue Type ahead API to support hyphens

### PR DESCRIPTION
**Pull request by ishratt**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-5662

### Change description ###
Modified current regex pattern to accept hyphens in the search string

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by AdilOoze — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringFunctionalTest.java [hunk 1]
```diff
@@ -59,6 +59,29 @@ void shouldRetrieveCourtVenues_By_SearchString_WithStatusCode_200() {
...
+        final var response = (LrdCourtVenueResponse[]) lrdApiClient.retrieveResponseForGivenRequest(HttpStatus.OK,
+                          "?search-string=Stoke-on", LrdCourtVenueResponse[].class, path);
+        assertThat(response).isNotEmpty();
+
+        var courtVenueResponse = new ArrayList<>(Arrays.asList(response));
+        var courtNameVerified = courtVenueResponse
+            .stream()
+            .filter(venue -> venue.getCourtName().strip().toLowerCase().contains("Stoke-".toLowerCase())
+                || venue.getSiteName().strip().toLowerCase().contains("Stoke-".toLowerCase())
+                || venue.getCourtAddress().strip().toLowerCase().contains("Stoke-".toLowerCase())
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringFunctionalTest.java:74]: why these filters are required inside test method ?

File: src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringIntegrationTest.java [hunk 2]
```diff
@@ -60,7 +76,8 @@ void shouldRetrieveCourtVenues_For_SearchString_And_CourtTypeId_WithStatusCode_2
     }
 
     @ParameterizedTest
-    @ValueSource(strings = {"?search-string=zz&court-type-id=1000", "?search-string=AB$&court-type-id=1000",
+    @ValueSource(strings = {"?search-string=abc--", "?search-string=ab__c", "?search-string=___c",
+        "?search-string=zz&court-type-id=1000", "?search-string=AB$&court-type-id=1000",
         "?search-string=AB$&court-type-id=1,2,$,4"})
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringIntegrationTest.java:81]: can you add other search strings to this test case with only repeated special chars like 3 underscores and likewise other all allowed special chars, please ?

File: src/test/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdCourtVenueControllerTest.java [hunk 3]
```diff
@@ -154,4 +175,10 @@ void testGetCourtVenuesBySearchStringWithInvalidCourtTypeIdWithComma() {
         assertThrows(InvalidRequestException.class, () ->
             lrdCourtVenueController.retrieveCourtVenuesBySearchString("ABC", ",1,2,", null, null, null, null));
     }
+
+    @Test
+    void testGetCourtVenuesBySearchStringWithConsecutiveSpecialCharacters() {
```
Comment (rajkatla-hmcts) [inline src/test/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdCourtVenueControllerTest.java:180]: can you add other search strings with only repeated special chars like 3 underscores and likewise other all allowed special chars, please ?

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringIntegrationTest.java (hunk 2 above)
Comment (ishratt) [reply src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringIntegrationTest.java:81]: sure, I'll add them.

File: src/test/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdCourtVenueControllerTest.java (hunk 3 above)
Comment (ishratt) [reply src/test/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdCourtVenueControllerTest.java:180]: sure, I'll add them.

File: src/functionalTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringFunctionalTest.java (hunk 1 above)
Comment (ishratt) [reply src/functionalTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringFunctionalTest.java:74]: @rajkatla-hmcts it is to check if each of the court venues returned in the response have one of those fields matching with the search string given. Currently when querying the db we look for the search string in these fields as well: `siteName`, `courtName`, `postcode` and `courtAddress` as you can see here https://github.com/hmcts/rd-location-ref-api/blob/07162e08b238f865302113c884f1c44ba929ddd7/src/main/java/uk/gov/hmcts/reform/lrdapi/repository/CourtVenueRepository.java#L50-L53

File: src/test/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdCourtVenueControllerTest.java (hunk 3 above)
Comment (ishratt) [reply src/test/java/uk/gov/hmcts/reform/lrdapi/controllers/LrdCourtVenueControllerTest.java]: addressed here [7a3ac72](https://github.com/hmcts/rd-location-ref-api/pull/473/commits/7a3ac72a8c9d5afffdb33a70158fc51a14a52465)

File: src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringIntegrationTest.java (hunk 2 above)
Comment (ishratt) [reply src/integrationTest/java/uk/gov/hmcts/reform/lrdapi/RetrieveCourtVenuesBySearchStringIntegrationTest.java:83]: addressed [7a3ac72](https://github.com/hmcts/rd-location-ref-api/pull/473/commits/7a3ac72a8c9d5afffdb33a70158fc51a14a52465)

**Review by rajkatla-hmcts — APPROVED** [review]

**Pull request merged**